src/main/java/com/minecolonies/api/items/ModToolTypes.java

```diff
@@ -0,0 +1,283 @@
+package com.minecolonies.api.items;
+
+import com.minecolonies.api.compatibility.Compatibility;
+import com.minecolonies.api.items.registry.ToolTypeEntry;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.api.util.constant.translation.ToolTranslationConstants;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.item.*;
+import net.minecraftforge.common.ToolAction;
+import net.minecraftforge.common.ToolActions;
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Supplier;
+
+/**
+ * Class used for storing and registering any ToolTypes.
+ */
+public class ModToolTypes
+{
+    /**
+     * The Forge registry for tool types. Do not directly add to this registry. Prefer using the
+     * register() function provided here.
+     */
+    private final static DeferredRegister<ToolTypeEntry> DEFERRED_REGISTER = DeferredRegister.create(new ResourceLocation(Constants.MOD_ID, "tooltypes"), Constants.MOD_ID);
+
+    public static String NONE_ID            = "none";
```

Comment:
Can you please inline these into registry methods? and ideally follow style (for mod[type]s.java) set in #10170
includes: member visibility, final, etc.

Reply:
Based on conversations in Discord, will keep as is with additional finals.